ports/psoc6/mplogger.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2013, 2014 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+// function to pipe debug messages to console in LOGGER DEBUG MODE
+void mplogger_print(char *message, ...) {
```

Comment:
This still is a function call. A macro that is empty if the MICROPY_LOGGER_DEBUG flag is not set would be better.

Reply:
I think you mean that the macro should be switched by a flag that is not defined at all (not equal to 0) when inactive? 

If yes, then we cannot switch it inside the `mpconfigport.h` since:
```
#define MICROPY_LOGGER_DEBUG                    (0)
```
would still mean that the flag is defined but equal to 0. Then we only have the Makefile option to activate/deactivate the debug logger messages. Will this be ideal since MPY generally uses switches in `mpconfigport.h`? Is there any other way to switch the macro with just `#ifdef` instead of `#if flag == value`

Reply:
What I meant : in order not to influence a non-debug version of the code the #ifdef should be used to either define a log routine that does something in debug mode or really leads to no code at all in the release version, i.e. not even an empty function call. In the latter case a macro avaluation the leads to no code and eats up the passed parameter.

Reply:
As discussed, the approach in this link with the do-while loop has been implemented. This approach is still expensive for the case when DEBUG mode is deactivated. The approach suggested in the link by _CodeLurker_ is similar to the in-built debug print infrastructure of MPY which we can later use (once the errors are resolved). 

Link: https://stackoverflow.com/questions/1644868/define-macro-for-debug-printing-in-c